### DISCUSSION TITLE
curiosity

### DISCUSSION THREAD

**Discussion by DhaosEsedess** [WhitewaterFoundry/Fedora-Remix-for-WSL#225] [General]
why is it that after install-desktop.sh is run running a program form the terminal like firefox causes it to fall back to x11 mode? is that intended? and if so is there any way to use Wayland when not in the RDP desktop?

**Comment by crramirez** [comment]
Hello,

The install-desktop script enables systemd, and there is a problem with firefox, systemd and wayland that makes it fallback to x11. Moreover when you start an RDP session, Fedora Remix on purpose disables wayland to avoid the applications "escape" the RDP window, removing the file socket `/run/user/1000/wayland-0`. It get restored after a cold restart of Fedora Remix.

In summary do the following to get firefox running again with wayland

1. Terminate fedora with `wsl --terminate fedoraremix`
2. Make the socket available to firefox even on systemd: `ln -sf /mnt/wslg/runtime-dir/wayland-0 /run/user/1000/wayland-0`

Tell me if it solves your problem
